Document HTTP redirections follow options

guillempages opened the pull request:
Update documentation according to PR https://github.com/esphome/esphome/pull/3100

## Description:
Add force_redirect and redirect_limit options

**Related issue (if applicable):**

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#3100

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

guillempages committed:
Update http_request.rst

Update documentation according to PR https://github.com/esphome/esphome/pull/3100

oxan reviewed (changes requested):
Can you rebase this upon the `next` branch? Right now your PR contains all changes from the `current` branch as well, which makes it hard to review.

guillempages commented:
Oops. Sorry :-(
I did that from the UI, and saw that I was pushing to the wrong branch, so I changed that, but I missed the fact that I had already branched off the wrong branch.
Fixed now.

guillempages commented on components/http_request.rst:
```diff
@@ -22,6 +22,8 @@ Configuration variables:
 - **timeout** (*Optional*, :ref:`config-time`): Timeout for request. Defaults to ``5s``.
 - **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+- **force_redirects** (*Optional*, boolean): Enable HTTP forced redirection. Defaults to ``true``.
+- **redirect_limit** (*Optional*, integer): Maximum amount of redirects to follow when enabled.
```
```suggestion
- **redirect_limit** (*Optional*, integer): Maximum amount of redirects to follow when enabled. Defaults to ``3``.
```

guillempages committed:
Add default value to redirect_limit

oxan commented on components/http_request.rst:
```diff
@@ -22,6 +22,8 @@ Configuration variables:
 - **useragent** (*Optional*, string): User-Agent header for requests. Defaults to ``ESPHome``.
 - **timeout** (*Optional*, :ref:`config-time`): Timeout for request. Defaults to ``5s``.
 - **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+- **force_redirects** (*Optional*, boolean): Enable HTTP forced redirection. Defaults to ``true``.
```
Force -> follow:
```suggestion
- **follow_redirects** (*Optional*, boolean): Enable following HTTP redirects. Defaults to ``true``.
```

guillempages replied:
(facepalm)

oxan reviewed (changes requested):
No worries!

guillempages committed:
Update components/http_request.rst

Co-authored-by: Oxan van Leeuwen <oxan@oxanvanleeuwen.nl>
